src/sync/sync_session.cpp

```diff
@@ -114,106 +97,17 @@ struct SyncSession::State {
     // PRECONDITION: the session state lock must be held at the time this method is called, until after it returns.
     virtual void wait_for_completion(SyncSession&, _impl::SyncProgressNotifier::NotifierType) const { }
 
-    virtual void override_server(std::unique_lock<std::mutex>&, SyncSession&, std::string, int) const { }
-
-    static const State& waiting_for_access_token;
     static const State& active;
     static const State& dying;
     static const State& inactive;
 };
 
-struct sync_session_states::WaitingForAccessToken : public SyncSession::State {
+struct sync_session_states::Active : public SyncSession::State {
     void enter_state(std::unique_lock<std::mutex>&, SyncSession& session) const override
-    {
-        session.m_deferred_close = false;
-    }
-
-    void refresh_access_token(std::unique_lock<std::mutex>& lock, SyncSession& session,
-                              std::string access_token,
-                              const util::Optional<std::string>& server_url) const override
     {
         session.create_sync_session();
+        session.m_session->bind();
```

Comment:
In Java this crashes with `../src/realm/sync/client.cpp:1569: [realm-core-10.0.0-alpha.4] Assertion failed: !m_initiated`

Looking at the Sync code, I suspect it is because bind is not idempotent, calling it multiple times is a logic error :/

Reply:
That's surprising - this is called immediately after a brand new sync session is created. Effectively this means that `bind()` should never be called on the same sync session. Do you have a full stack trace for this?

Reply:
I did have a full stack trace, but it didn't show anything interesting except the above failure. When using the debugger it looked like the Active State was entered multiple times, which makes sense given that it would trigger the error. I don't know why that is happening though.

Reply:
I will look into why that's the case - probably a bug on my part when I removed the `WaitingForAccessToken` state.